src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -170,41 +207,68 @@ private TFSession LoadTFSession(byte[] modelBytes)
             catch (Exception ex)
             {
 #pragma warning disable MSML_NoMessagesForLoadContext
-                throw _host.ExceptDecode(ex, "Tensorflow exception triggered while loading model.");
+                throw env.ExceptDecode(ex, "Tensorflow exception triggered while loading model.");
 #pragma warning restore MSML_NoMessagesForLoadContext
             }
             return new TFSession(graph);
         }
 
+        private static TFSession LoadTFSession(IHostEnvironment env, string exportDirSavedModel)
+        {
+            env.CheckValue(exportDirSavedModel, nameof(exportDirSavedModel));
+            var sessionOptions = new TFSessionOptions();
+            var exportDir = exportDirSavedModel;
+            var tags = new string[] { "serve" };
+            var graph = new TFGraph();
+            var metaGraphDef = new TFBuffer();
+
+            return TFSession.FromSavedModel(sessionOptions, null, exportDir, tags, graph, metaGraphDef);
+        }
+
+        private static TFSession GetSession(IHostEnvironment env, string modelPath)
+        {
+            var isFrozen = TensorFlowUtils.IsFrozenTensorFlowModel(modelPath);
+            if (isFrozen)
+            {
+                byte[] modelBytes = CheckFileAndRead(env, modelPath);
+                return LoadTFSession(env, modelBytes);
+            }
+            else
+                return LoadTFSession(env, modelPath);
+        }
+
         private static byte[] CheckFileAndRead(IHostEnvironment env, string modelFile)
         {
             env.CheckNonWhiteSpace(modelFile, nameof(modelFile));
             env.CheckUserArg(File.Exists(modelFile), nameof(modelFile));
             return File.ReadAllBytes(modelFile);
         }
 
-        public TensorFlowTransform(IHostEnvironment env, string modelFile, string[] inputs, string[] outputs) :
-            this(env, CheckFileAndRead(env, modelFile), inputs, outputs)
+        public TensorFlowTransform(IHostEnvironment env, string model, string[] inputs, string[] outputs, bool isTemporaryModelPath) :
+            this(env, GetSession(env, model), inputs, outputs, true)
```

Comment:
>true [](start = 63, length = 4)

Is this correct?

Reply:
I was passing default true into the base constructor to match its signature. Line #251  overrides the IsFrozen field 

---
In reply to: [219207918](https://github.com/dotnet/machinelearning/pull/853#discussion_r219207918) [](ancestors = 219207918)

Reply:
refactored code so now we have only 1 constructor

---
In reply to: [219228320](https://github.com/dotnet/machinelearning/pull/853#discussion_r219228320) [](ancestors = 219228320,219207918)